iceoryx_utils/include/iceoryx_utils/cxx/forward_list.hpp

```diff
@@ -0,0 +1,406 @@
+    bool emplace_front(ConstructorArgs&&... args) noexcept;
+
+    /// @brief construct element inplace at begining of list
+    /// @param[in] args T-typed construction parameters (initializer list)
+    /// @param[in] afterToBeEmplacedIter position in list to (construct)insert after
+    /// @return iterator to the newly added element
+    template <typename... ConstructorArgs>
+    iterator emplace_after(const_iterator afterToBeEmplacedIter, ConstructorArgs&&... args) noexcept;
+
+    /// @brief insert element after iterator position
+    /// @param[in] citer iterator with the position to insert after
+    /// @param[in] data reference to element to add
+    /// @return iterator to the newly added element
+    iterator insert_after(const_iterator citer, const T& data) noexcept;
+
+    /// @brief construct element inplace at begining of list
+    /// @param[in] citer iterator with the position to insert after
+    /// @param[in] data universal reference perfectly forwarded to client class
+    /// @return iterator to the newly added element
+    iterator insert_after(const_iterator citer, T&& data) noexcept;
+
+  private:
+    struct ListNode
+    {
+        T data;
+        size_type nextIdx;
+    };
+
+    void init() noexcept;
+    ListNode* getNodePtrFromIdx(size_type) const noexcept;
+    ListNode* getHeadOfFreeElementList() const noexcept;
+    ListNode* getNodePtr() const noexcept;
+
+    bool isValidIteratorIndex(size_type index) const noexcept;
+    bool isValidElementIndex(size_type index) const noexcept;
+    static void errorMessage(const char* f_source, const char* f_msg) noexcept;
+
+
+    //***************************************
+    //    members
+    //***************************************
+
+    // two extra slots in the list to handle the 'before_begin' element of each:(internal) freeList and usedList
+    // the necessity for 'before_begin' elements stems from the way a forward_list removes elements at an arbitrary
+    // position. Removing the front-most list element (aka begin()) requires an element pointing towards this position,
+    // hence 'before_begin'. The before_begin index is the head of each free-slots or used-slots list respectively.
+    static constexpr size_type INTERNAL_CAPACITY{size_type(Capacity) + 2U};
+    static constexpr size_type BEFORE_BEGIN_USED_INDEX{0U};
+    static constexpr size_type BEFORE_BEGIN_FREE_INDEX{1U};
+    static constexpr size_type INVALID_INDEX{size_type(Capacity) + 2U};
+
+    using element_t = uint8_t[sizeof(ListNode)];
+    alignas(alignof(T)) element_t m_data[INTERNAL_CAPACITY];
+
+    size_type m_size{0u};
+}; // forward_list
+
+// template <typename IterType, typename IterTypeOther>
+// bool operator==<typename std::enable_if<is_forward_list_iterator<IterType>::value>::type,
+//                 typename std::enable_if<is_forward_list_iterator<IterTypeOther>::value>::type>(
+//     const IterType& lhs_iter, const IterTypeOther& rhs_iter) noexcept;
```

Comment:
Remove?

Reply:
uups, sure! removed in 6bbaaffef5d63b6e7ff0785d850bb02c93ba638b